ql/cashflows/overnightindexedcoupon.cpp

```diff
@@ -54,8 +54,7 @@ namespace QuantLib {
                 Real compoundFactor = 1.0;
 
                 // already fixed part
-                Date today = Settings::instance().evaluationDate();
-                while (i<n && fixingDates[i]<today) {
```

Comment:
No, it's not `today` that should be replaced.  Knowing today's date tells us whether the fixing for a given date should be retrieved from history (before today) or forecast off the curve (after today).  This shouldn't change, or we'll end up either trying to forecast a fixing in the past or trying to retrieve a future fixing from history, both of which will raise an exception.

What should change is that instead of looping over all of `fixingDates` (that is, between the accrual start date and end date) we should compound fixings only between the accrual start date and the passed accrual date.  The case corresponding to the current `rate()` would pass the accrual end date.

Reply:
I see....

Reply:
i'll upload my second attempt.